flavors/publisher_test.go

```diff
@@ -0,0 +1,182 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package flavors
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/stretchr/testify/mock"
+	"go.uber.org/goleak"
+
+	"github.com/elastic/cloudbeat/resources/utils/testhelper"
+)
+
+func TestPublisher_HandleEvents(t *testing.T) {
+	type testCase struct {
+		name              string
+		interval          time.Duration
+		threshold         int
+		ctxTimeout        time.Duration
+		eventCount        int
+		expectedEventSize []int
+		closeChannel      bool
+	}
+	testCases := []testCase{
+		{
+			name:              "Publish events on threshold reached",
+			interval:          time.Minute,
+			threshold:         1,
+			ctxTimeout:        200 * time.Millisecond,
+			eventCount:        2,
+			expectedEventSize: []int{1},
+		},
+		{
+			name:              "Publish events on threshold reached",
```

Comment:
Why have two tests with the same purpose? 
Either update the name or delete the test